### PR TITLE
fix(filtering): handle multiple events firing for column of type date #10233

### LINKED ISSUE #10233: IgxGrid filtering event fires multiple times for column of type 'date'
## Description  
The filtering (and correspondingly  filteringDone) event fires multiple times for a date column.

 * igniteui-angular version: 11.1.26, 12.1.x
 * browser: any

## Steps to reproduce  

1. Open the following [sample](https://stackblitz.com/edit/filtering-12-1-1?file=src/app/grid/grid-filtering-sample/grid-filtering-sample.component.html)
2. Open the browser console
3. Click the quick filter button of the "Order Date" column
4. Select any date from the calendar
5. Click anywhere to lose focus of the filled-in date input / Click the "Close" button

## Result  
The filtering event is fired on the following occasions:
1. Once the  quick filter button of the "Order Date" column is clicked _when no filter is applied_
2. Twice when selecting a date from the calendar
3. Once when losing focus of the filled-in date input

## Expected result  
The filtering event to be fired as for the other columns, f.ex. of type 'string'. To be fired only once when selecting a date from the calendar.

### PR DESCRIPTION
**Pull request by zhosti**
Closes #10233 

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts [hunk 1]
```diff
@@ -6460,3 +6494,85 @@ const verifyChipVisibility = (fix, index: number, shouldBeFullyVisible: boolean)
...
+    const reset = filterUiRow.queryAll(By.css('button'))[0];
+    expect(reset.nativeElement.classList.contains('igx-button--disabled')).toEqual(true);
+};
+
+const emitFilteringDoneOnInputClear = (fix, grid, filterVal, columnName, condition) => {
+    filterGrid(fix, grid, columnName, filterVal, condition);
+
+    GridFunctions.clickFilterCellChip(fix, columnName);
+
+    spyOn(grid.filteringDone, 'emit');
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts:6520]: Isn't this spy a duplicate of the one in beforeEach?

File: projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts [hunk 2]
```diff
@@ -2794,6 +2703,131 @@ describe('IgxGrid - Filtering Row UI actions #grid', () => {
...
+            emitFilteringDoneOnResetClick(
+                fix,
+                grid,
+                'search',
+                'ProductName',
+                IgxStringFilteringOperand.instance().condition('contains')
+            );
+        }));
+
+        it('Should emit filteringDone when we clicked reset - Number column type', fakeAsync(() => {
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts:2743]: Can we expand these tests to Boolean and Currency columns (not sure if we have some other type as well - Time probably?)?

**Review by gedinakova — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts (hunk 1 above)
Comment (zhosti) [reply projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts:6520]: Actually yes, it is duplicated. So I will remove it.

File: projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts (hunk 2 above)
Comment (zhosti) [reply projects/igniteui-angular/src/lib/grids/grid/grid-filtering-ui.spec.ts:2743]: Yes, we can. I will add more columns in grid sample data with the (boolean, currency and time) column types.

**Review by gedinakova — APPROVED** [review]

**Pull request merged**